.github/workflows/ui-tests.yml

```diff
@@ -0,0 +1,86 @@
+name: UI Tests
+
+on: [push, pull_request]
+
+jobs:
+  ui-test:
+    name: Visual Regression
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        group: [ui_test]
+        python: [3.8]
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python }}
+
+    - name: Set up Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+
+    - name: Cache pip on Linux
+      uses: actions/cache@v2
+      if: startsWith(runner.os, 'Linux')
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ matrix.python }}-${{ hashFiles('**/requirements.txt', 'setup.py') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{ matrix.python }}
+
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+    - name: Cache yarn
+      uses: actions/cache@v2
+      id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    - name: Install dependencies
+      env:
+        GROUP: ${{ matrix.group }}
+      run: |
+        bash ./scripts/ci_install.sh
```

Comment:
At some point later, we could also extend `ci_install` to install specific dependencies for the `ui_tests` group, for example `xeus-python` to enable the debugger, similar to:

https://github.com/jupyterlab/jupyterlab/blob/9b5a6fd63df142424ec785e07993c8586cd7107c/scripts/ci_install.sh#L53-L56